conda_smithy/templates/circle.yml.tmpl

```diff
@@ -31,7 +31,11 @@ jobs:
         DOCKER_IMAGE: "{{ data.config["docker_image"][-1] }}"
 {%- endif %}
     steps:
+{%- if clone_depth %}
+      - run: git clone --depth {{ clone_depth }} "$CIRCLE_REPOSITORY_URL" --branch "$CIRCLE_BRANCH"
+{%- else %}
       - checkout
+{%- endif %}
```

Comment:
Is this going to break for repos with special access permissions needed?

Reply:
Oh quite possibly. This one was just pulled from some docs and I have no idea if it's worth implementing. The thing I definitely found out is that Circle doesn't let you customize the checkout step. You just have to implement it yourself if you want to do anything custom.

Reply:
Looks like it breaks, period: https://app.circleci.com/pipelines/github/conda-forge/cf-autotick-bot-test-package-feedstock/1300/workflows/28077417-7245-4308-86c7-8d8ff74ab4ef/jobs/1158

I suppose I could try to dig into this and figure out a clone command that works, or I could just leave it out. I'm inclined to the latter, and I doubt people will miss having the ability to set a finite clone depth on Circle.

Reply:
I'm ok removing the circle code. FWIW, it appears the branch circle gives in the variable is not the one in the repo.